onnxruntime/core/optimizer/dynamic_quantize_matmul_fusion.cc

```diff
@@ -12,6 +12,27 @@ using namespace ONNX_NAMESPACE;
 using namespace ::onnxruntime::common;
 namespace onnxruntime {
 
+inline int64_t GetLastDimOfTensorShapeProto(const TensorShapeProto& shape) {
+  return shape.dim(shape.dim_size() - 1).dim_value();
+}
```

Comment:
Any chance you would see an empty shape (scalar) or something that doesn't have dim_value set here?